### PR TITLE
Nodejs binary module mappings

### PR DESCRIPTION
**Pull request by gregmagolan**
* fix(builtin): nodejs_binary collects module_mappings for linker

Module name cannot be equal to the workspace name with the linker so this change removes the regression test for 834. This was a corner case and was already broken with the 1.0 for npm_package_bin so this fix makes nodejs_binary consistent with npm_package_bin and the new desired behavior for 1.0.

* fix(builtin): legacy module_mappings_runtime_aspect handles dep with module_name but no module_root

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature (please, look at the "Scope of the project" section in the README.md file)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: internal/node/node.bzl [hunk 1]
```diff
@@ -277,7 +277,7 @@ _NODEJS_EXECUTABLE_ATTRS = {
     "data": attr.label_list(
         doc = """Runtime dependencies which may be loaded during execution.""",
         allow_files = True,
-        aspects = [node_modules_aspect, module_mappings_runtime_aspect],
+        aspects = [node_modules_aspect, module_mappings_aspect, module_mappings_runtime_aspect],
```
Comment (alexeagle) [inline internal/node/node.bzl]: I was hoping to remove the module_mappings_runtime_aspect at the same time. This is a perf regression since we'll walk the deps two separate times, gathering the same data and returning it in slightly different format.
I guess we can do that as follow-up but ideally we should do it now? Soon?
